### PR TITLE
Feature/redux setup

### PR DESCRIPTION
**Pull request by fawadmehmood1**
# Description

Please include a summary of the changes and the related issue.

- Added Redux Store
- Added Auth Api Slice

## Type of change

Please select type of issue addressed,

- [ ] Bug fix (Mention Ticket numbers)

- [ ] New feature ( Story Number ) 

# Dependences / Libraries Installed

Please list the packages and libraries installed (if any)

- [x] ( @reduxjs/toolkit )  (https://www.npmjs.com/package/@reduxjs/toolkit) 
- [x] ( react-redux )  (https://www.npmjs.com/package/react-redux) 
- [x] (redux-persist) (https://www.npmjs.com/package/redux-persist)
- [x] (cookies-next) (https://www.npmjs.com/package/cookies-next)

# How Has This Been Tested By The Developer?
- Verified if Redux is setup properly

# Checklist:

- [x] I have performed a self review of my code.

- [x] I have removed consoles, debuggers and unnecessary code blocks.

- [x] My code follows the style guidelines of this project

- [x] My changes generate no new warnings or errors

- [ ] I have commented my code, particularly in hard-to-understand areas

- [ ] New and existing unit tests pass locally with my changes

- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I understand that I am accountable for any code added in this PR, and any negative impact will effect my performance evaluation

### CODE REVIEW

File: constants/index.ts [hunk 1]
```diff
@@ -21,3 +21,5 @@ export const PROTECTED_ROUTES = [
 ];
 
 export const AUTH_ROUTES = [ROUTES.SIGN_UP, ROUTES.SIGN_IN];
+
+export const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL;
```
Comment (imrankhan-stixor) [inline constants/index.ts]: As there will be multiple urls due to microservices so rename the url for the purpose it's related to

File: redux/apis/auth-api.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { getCookie } from "cookies-next";
+
+export const authApi = createApi({
+  reducerPath: "authApi",
+  baseQuery: fetchBaseQuery({
+    baseUrl: API_BASE_URL,
+    prepareHeaders: (headers) => {
+      const token = getCookie("token");
+
+      // TODO: If token is expired logout and reset api states.
```
Comment (imrankhan-stixor) [inline redux/apis/auth-api.ts]: remove this comment if not needed

File: redux/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
+import type { RootState, AppDispatch } from "./store";
+
+// Use throughout your app instead of plain `useDispatch` and `useSelector`
```
Comment (imrankhan-stixor) [inline redux/hooks.ts:4]: remove if it's not necessarily needed

File: types/index.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export interface UserState {
+  user?: {
+    id?: number;
```
Comment (imrankhan-stixor) [inline types/index.ts]: change this to string

**Review by imrankhan-stixor — CHANGES_REQUESTED** [review]

**Review by hassantahir767 — APPROVED** [review]

File: redux/hooks.ts (hunk 3 above)
Comment (fawadmehmood1) [reply redux/hooks.ts:4]: @imrankhan-stixor , it's a highlight comment, it should be there.

**Review by imrankhan-stixor — APPROVED** [review]

**Pull request merged**